core/internal/runmetric/runconfigmetrics.go

```diff
@@ -74,6 +95,9 @@ func (rcm *RunConfigMetrics) encodeToRunConfigData(
 			rcm.handler.definedMetrics[metric.Step],
 			encodedMetrics,
 			indexByName,
+			// operating under the assumption that the step metric is
+			// never a glob metric.
```

Comment:
Please capitalize the first letter for comments.

What happens if this assumption is broken? Do we need to handle that?

Reply:
> i don't think we allow the step to be a glob currently

Is it not an assumption then? Can the comment be

```
// Step metrics are never interpreted as globs.
```